src/script/common/c_content.cpp

```diff
@@ -128,6 +128,77 @@ void push_item_definition(lua_State *L, const ItemDefinition &i)
 	lua_setfield(L, -2, "description");
 }
 
+void push_item_definition_full(lua_State *L, const ItemDefinition &i)
+{
+	std::string type;
+	// This code is pure garbage, but that is what reviewers are for!
+	const bool use_caps = i.type == ITEM_TOOL;
+
+	switch (i.type)
+	{
+		case ITEM_NONE:
+		{
+			type = "none";
+			break;
+		}
+		case ITEM_NODE:
+		{
+			type = "node";
+			break;
+		}
+		case ITEM_CRAFT:
+		{
+			type = "craftitem";
+			break;
+		}
+		case ITEM_TOOL:
+		{
+			type = "tool";
+			break;
+		}
```

Comment:
you don't handle the default case this is a problem. you should trigger a FATAL_ERROR